OpenMeteo/fetch_OpenMeteo_data.py

```diff
@@ -0,0 +1,120 @@
+import openmeteo_requests  # Importing required libraries
+import pandas as pd
+import requests_cache
+from retry_requests import retry
+
+
+class WeatherDataFetcher:
```

Comment:
Could this be changed the following, so that the name is more descriptive of the data it is getting
```suggestion
class OpenMeteoWeatherDataFetcher:
```